src/aiida/engine/processes/functions.py

```diff
@@ -373,17 +373,21 @@ def build(func: FunctionType, node_class: t.Type['ProcessNode']) -> t.Type['Func
             LOGGER.warning(f'function `{func.__name__}` has invalid type hints: {exception}')
             annotations = {}
 
-        try:
-            parsed_docstring = docstring_parser.parse(func.__doc__)
-        except Exception as exception:
-            LOGGER.warning(f'function `{func.__name__}` has a docstring that could not be parsed: {exception}')
-            param_help_string = {}
+        if func.__doc__ is None:
```

Comment:
This fixes:

```
src/aiida/engine/processes/functions.py:377: error: Argument 1 to "parse" has incompatible type "str | None"; expected "str"  [arg-type]
```

Github is very confused, the diff here is actually not that bad, we're just handling the `func.__doc__ is None` case pre-emptively. (We could also log a warning if a calcfunction is missing a docstring, but that seems overly pedantic)

Reply:
> We could also log a warning if a calcfunction is missing a docstring, but that seems overly pedantic

Yes, I think that is a bit too much, I won't throw warning for that.